lib/logstash/codecs/cloudwatch_logs.rb

```diff
@@ -25,11 +31,13 @@ def decompress(data)
   end
 
   def parse(json, &block)
-    base = json.reject { |k,_| k == "logEvents" }.freeze
-    events = json["logEvents"]
+    events = json.delete("logEvents")
+    json.freeze
 
     events.each do |event|
-      yield LogStash::Event.new(base.merge(event))
+      epochmillis = event.delete("timestamp").to_i
+      event[LogStash::Event::TIMESTAMP] = LogStash::Timestamp.at(epochmillis / 1000, (epochmillis % 1000) * 1000)
```

Comment:
@threadwaste It looks like the awslogs driver does set the Logs event timestamp to the timestamp [of the Docker native event](https://github.com/docker/docker/blob/b248de7e332b6e67b08a8981f68060e6ae629ccf/daemon/logger/awslogs/cloudwatchlogs.go#L268). 

I'm not sure what would cause the Docker timetamp to not match the timestamp embedded in your event message, though. Might be worth looking into?